flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -220,4 +220,10 @@ boolean registerListener(NotificationListener listener) {
 			throw new IllegalStateException("Already registered listener.");
 		}
 	}
+
+	public int getNumberOfQueuedBuffers() {
+		synchronized (buffers) {
```

Comment:
Can we avoid synchronization here? The metrics should never influence (block) the other code.
I would rather have the metrics be one off once in a while and avoid the lock here.